client2/src/game/Game.js

```diff
@@ -548,35 +575,12 @@ class Game extends Phaser.Scene {
         this.keyboardKeys.d.on("up", this.moveRightReleased, this);
 
         this.keyboardKeys.enterChat.on("down", () => {
-            if (this.player.hitPoints <= 0) {
-                // Close the box. Can't chat while dead.
-                this.GUI.chatInput.isActive = false;
-                this.GUI.chatInput.style.visibility = "hidden";
-                this.GUI.chatInput.blur();
-                this.GUI.chatInput.value = "";
-                return;
-            }
-            // Check if the chat input box is open.
-            if (this.GUI.chatInput.isActive === true) {
-                // Close the box, and submit the message.
-                this.GUI.chatInput.isActive = false;
-                this.GUI.chatInput.style.visibility = "hidden";
-                this.GUI.chatInput.blur();
-
-                // Don't bother sending empty messages.
-                if (this.GUI.chatInput.value !== "") {
-                    // Send the message to the server.
-                    window.ws.sendEvent("chat", this.GUI.chatInput.value);
-
-                    // Empty the contents ready for the next chat.
-                    this.GUI.chatInput.value = "";
-                }
-            }
-            else { // Not open, so open it.
-                this.GUI.chatInput.isActive = true;
-                this.GUI.chatInput.style.visibility = "visible";
-                this.GUI.chatInput.focus();
-            }
+            // Send player hitPoints and websocket to be used by chatInput
+            // player is undefined 'cause it's not defined,
+            PubSub.publish(ENTER_KEY, {
+                hitPoints: this.player.hitPoints,
+                webSocket: window.ws,
```

Comment:
The websocket connection is still available globally, so even in the event handler that picks up this event it can still use `window.ws`, so this `webSocket` property isn't needed.

Reply:
Hmm the reason behind this is I'm not fan of using the `window` object, but in this case it perfectly makes senses.

Here's some reason why:
1. I'm worried about is that a library that will be imported in the future will also use `window.ws` hence a collision.
2. We have to make sure that window.ws is defined at the top before everything referencing to it was called or it will be undefined.

And here's some possible solutions:
1. since `window.ws` is such a common word we could change it to a more unique one such as `window.dungeonZWS`.
2. well just make sure that `window.ws` is always the first one being called before reference calls.

Reply:
Or a state could be added for it, maybe as part of the Application https://github.com/Arcanorum/dungeonz/blob/feat/react/client2/src/shared/state/Application.js

I think I just liked using window.ws so far as it is used in a lot of places and I previously couldn't be bothered to import it every time, but it probably should be kept somewhere better now. It would just be a lot of existing uses to tidy up.

Reply:
Refactored at 01ceb8d